cmd/create/oidcprovider/cmd.go

```diff
@@ -265,3 +277,30 @@ func sha1Hash(data []byte) string {
 	hashed := hasher.Sum(nil)
 	return hex.EncodeToString(hashed)
 }
+
+func validateOperatorRoles(reporter *rprtr.Object, awsClient aws.Client, cluster *cmv1.Cluster) ([]string, error) {
+	var missingRoles []string
+
+	operatorIAMRoles := cluster.AWS().STS().OperatorIAMRoles()
+
+	if len(operatorIAMRoles) == 0 {
+		return missingRoles, fmt.Errorf("No Operator IAM roles found for cluster %s", cluster.Name)
+	}
+
+	for _, operatorIAMRole := range operatorIAMRoles {
+		roleARN := operatorIAMRole.RoleARN()
+
+		roleName := strings.Split(roleARN, "/")[1]
+
+		exists, err := awsClient.CheckRoleExists(roleName)
+		if err != nil {
+			return missingRoles, err
+		}
+
+		if !exists {
+			missingRoles = append(missingRoles, roleName)
+		}
+	}
+
+	return missingRoles, nil
+}
```

Comment:
@vkareh This could go into a pkg either aws or ocm thoughts? I wasn't sure if it fitted either, maybe ocm?

Reply:
I'd leave it in the cmd, since it blurs the ocm vs aws line, it uses both, so let's leave it where it is